common/src/services/state.service.ts

```diff
@@ -1359,24 +1362,16 @@ export class StateService<TAccount extends Account = Account>
   }
 
   async getEnvironmentUrls(options?: StorageOptions): Promise<any> {
-    return (
-      (await this.getGlobals(this.reconcileOptions(options, await this.defaultOnDiskOptions())))
-        ?.environmentUrls ?? {
-        base: null,
-        api: null,
-        identity: null,
-        icons: null,
-        notifications: null,
-        events: null,
-        webVault: null,
-        keyConnector: null,
-        // TODO: this is a bug and we should use base instead for the server detail in the account switcher, otherwise self hosted urls will not show correctly
-        server: "bitwarden.com",
-      }
-    );
+    options = this.reconcileOptions(options, await this.defaultOnDiskOptions());
+    if (this.state.activeUserId == null) {
+      return (await this.getGlobals(options)).environmentUrls ?? new EnvironmentUrls();
+    }
+    return (await this.getAccount(options))?.settings?.environmentUrls ?? new EnvironmentUrls();
```

Comment:
Cool, this looks like it handles the case where directory connector has you set environment urls before logging in.

Reply:
Yep! Desktop works that way too, so you shouldn't need any special stuff to make this work.